Change sil viz to use show, add to params

mgarod opened the pull request:
Addresses Silhouette Visualizer for https://github.com/DistrictDataLabs/yellowbrick/issues/960

I have made the following changes:

1. Add `show` to the parameters of the quick method
2. Make the visualizer in the quick method use show instead of poof

Questions for the @DistrictDataLabs/team-oz-maintainers:

- [x] I didn't modify the unittest because it appeared to be sufficient as it stood. Here is the unit test that already existed before beginning my assignment https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_cluster/test_silhouette.py#L177-L189
- [x] Is there supposed to be additional documentation added other than within the module itself.

No picture provided because there has been no change to any output. Tests still pass.

bbengfort commented on yellowbrick/cluster/silhouette.py:
```diff
@@ -285,26 +285,31 @@ def silhouette_visualizer(
+    ax : matplotlib Axes, default: None
+        The axis to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
```
Thank you for re-arranging the docstring params to match the function signature!

mgarod commented:
Thanks for the comprehensive review!

- RE `random_state` and `legend`: I traced them and could not determine where they actually get used. They kept getting passed in kwargs and died. I removed them from the test.

- In the test I set `show=False` which removed the warnings from test report.

- Added `silhouette_visualizer` to the api reference.

- Simply for a starting point, I added to the silhouette.rst. I put a short description of the quick method, and a copy-paste of the previous nfl example. After building the docs I realize that it produces exact same picture again, so I set `show=False`. Should I create an entirely new example and use new data to show off the quick method? Maybe I don't need an example at all, and should just show the one-liner of the quick method?

pdamodaran commented:
Thanks @mgarod for addressing the comments @bbengfort made to this PR.  I like that you included a quickmethod example in the documentation.  I think you can keep the example that you included in the documentation but take out `show=False`.  I think it will be good to demonstrate that the same visualization can be rendered by implementing the quick method.  If you can include that information to the documentation, that will be great.

bbengfort commented on tests/test_cluster/test_silhouette.py:
```diff
@@ -183,7 +183,7 @@ def test_quick_method(self):
 
         model = MiniBatchKMeans(3, random_state=343)
-        oz = silhouette_visualizer(model, X, random_state=93, legend=False)
+        oz = silhouette_visualizer(model, X, show=False)
```
:+1:

pdamodaran replied:
@bbengfort - could you explain what you mean by `Models` and `Workflow` in your example above?

rebeccabilbro replied:
Hey there @pdamodaran! Hope things have been well! Thought I'd jump in here since I couldn't resist 🙂 

`Models`: The type of machine learning that the `Visualizer` is designed for (e.g. Classification, Regression, Clustering &mdash; akin to the "default task" categorization on the [UCI Machine Learning Repository](https://archive.ics.uci.edu/ml/datasets.php)). For example, what kind of machine learning should you expect to do with a `SilhouetteVisualizer`?

`Workflow`: Where does this `Visualizer` fit into the [machine learning workflow](https://www.scikit-yb.org/en/develop/about.html#model-selection) &mdash; is it used for feature analysis, model evaluation, or hyperparameter tuning? For instance, at what stage of the workflow would a `ClassificationReport` be used?

pdamodaran replied:
Thanks @rebeccabilbro for the explanation - it makes sense to me now.

@bbengfort - The template you suggested above looks good to me.

mgarod commented:
Added the "at-a-glance" table to the top of the documentation page with links to the api (though the link I had to create for the quick method doesn't exist yet).

Created a new example with the credit dataset for the quick method documentation.
